crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -125,6 +125,94 @@ impl Ellipse {
     }
 }
 
+/// A primitive shape formed by the region between two circles, also known as a ring.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[doc(alias = "Ring")]
+pub struct Annulus {
+    /// The inner radius of the annulus
+    pub inner_radius: f32,
+    /// The outer radius of the annulus
+    pub outer_radius: f32,
+}
+impl Primitive2d for Annulus {}
+
+impl Default for Annulus {
+    /// Returns the default [`Annulus`] with radii of `0.5` and `1.0`.
+    fn default() -> Self {
+        Self {
+            inner_radius: 0.5,
+            outer_radius: 1.0,
+        }
+    }
+}
+
+impl Annulus {
+    /// Create a new [`Annulus`] from an inner and outer radius
+    #[inline(always)]
+    pub const fn new(inner_radius: f32, outer_radius: f32) -> Self {
+        Self {
+            inner_radius,
+            outer_radius,
+        }
+    }
+
+    /// Get the diameter of the annulus
+    #[inline(always)]
+    pub fn diameter(&self) -> f32 {
+        2.0 * self.outer_radius
+    }
+
+    /// Get the width of the annulus
+    #[inline(always)]
+    pub fn width(&self) -> f32 {
+        self.outer_radius - self.inner_radius
+    }
+
+    /// Get the area of the annulus
+    #[inline(always)]
+    pub fn area(&self) -> f32 {
+        PI * (self.outer_radius.powi(2) - self.inner_radius.powi(2))
+    }
+
+    /// Get the perimeter or circumference of the annulus,
+    /// which is the sum of the outer and inner edges.
+    #[inline(always)]
+    #[doc(alias = "circumference")]
+    pub fn perimeter(&self) -> f32 {
+        2.0 * PI * (self.outer_radius + self.inner_radius)
+    }
+
+    /// Finds the point on the annulus that is closest to the given `point`:
+    ///
+    /// - If the point is outside the annulus and closer to the outer perimeter
+    /// (outside the annulus completely), the returned point will be on the outer edge of the perimeter.
+    /// - If the point is outside the annulus and closer to the inner perimeter
+    /// (inside the inner circle), the returned point will be on the inner edge of the perimeter.
+    /// - Otherwise, it will be inside the annulus and returned as is.
```

Comment:
Maybe something like this?
```suggestion
    /// - If the point is outside of the annulus completely, the returned point will be on the outer perimeter.
    /// - If the point is inside of the inner circle (hole) of the annulus, the returned point will be on the inner perimeter.
    /// - Otherwise, the returned point is overlapping the annulus and returned as is.
```

Reply:
I like this! Clear and less verbose.